src/DotNetWorker.Core/DotNetWorker.Core.csproj

```diff
@@ -19,16 +19,22 @@
     <Compile Remove="FunctionsDebugger.cs" />
   </ItemGroup>
 
-  <ItemGroup>
-    <PackageReference Include="Azure.Core" Version="1.10.0" />
-    <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="5.0.0" />
-    <PackageReference Include="System.Collections.Immutable" Version="5.0.0" />
-  </ItemGroup>
-
-  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <ItemGroup>
+        <PackageReference Include="Azure.Core" Version="1.10.0" />
+    </ItemGroup>
+    
+    <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
+      <PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.0" />
+      <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="6.0.0" />
+      <PackageReference Include="System.Collections.Immutable" Version="6.0.0" />
+   </ItemGroup>
+    
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == 'net5.0'">
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="5.0.0" />
     <PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.4" />
     <PackageReference Include="Microsoft.Extensions.Options" Version="5.0.0" />
+      <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
+      <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="5.0.0" />
```

Comment:
but why not use version `7.0.0`

Reply:
As an SDK we have several restrictions around what packages we can and cannot use, and also their versions. We have many competing customer requirements we need to balance, back compat / breaking change requirements, and general guidelines to follow.

1. This is multi-targeting, so we are matching the .NET version with the packages released around that time. When you target net6.0, you should get the 6.0.0 dependencies. We will probably only bring in 7.0.0 packages if we have a net7.0 target. This is more of a pattern core .NET packages and frameworks use.
2. Reving major versions of dependencies can be considered a breaking change and would require us to also rev our matching version to follow SemVer semantics.
3. As a library, it is actually beneficial for us to target the lowest dependency versions possible (provided it has the features we want). Customers can pin to a higher version themselves via their own package reference to the desired higher version. This lets us cast as wide of a compatibility net as possible. We do not want to _force_ an upgrade, but we also don't want to _block_ an upgrade. We want to give as much control over package versions to the customer as possible.

Reply:
@jviau 
Thank you for your explanation; I certainly grasp the intent and your concerns.

This said, providing a base packages, namely `DotNetWorker.Core`, `DotnetWorker`, `DotNetWorker.Grpc` can safely target `netstandard2.0`, `net6.0`, `net7.0` and be adjusted with some conditionals in the project to reference the proper libraries.

This can be also implemented in the tests projects that perform the same tests and verify the specifics versions equally, which (in my opinion) be much better than relying only on `netstandard2.0' and 'net5.0' that it will work with the officially supported TFM.

In this example however, `Microsoft.Extensions.Hosting` version **7.0.0** is fully backward compatible to `netstandard2.0`, `net6.0`, `net7.0` ; since itself support those frameworks, please refer to [Microsoft.Extensions.Hosting](https://www.nuget.org/packages/Microsoft.Extensions.Hosting).    So if the `.csproj` refer to this version and has multi-targeting activated and tests are successful, it is reasonable to think it will work with your users.

Please note that this is also and **SDK** that are concerns with the same as stated earlier.

This also applies to the SDK portion, generator.

Reply:
@bobbyangers those dependencies will be revised and updated in a new major version, but can you share more information about whether that is blocking any of your scenarios? 

Multi-targeting with conditional dependencies is already the approach followed by SDK, but this also has to be done with care to ensure there will be no experiences where we force a major dependency bump within a given major version of our packages.

Reply:
@fabiocav, of course, I understand.

I have proposed a separate PR (#1242) to handle multiple - simultaneous versions that would avoid address this concern.